megamek/src/megamek/common/Aero.java

```diff
@@ -1841,14 +1841,14 @@ public int calculateBattleValue(boolean ignoreC3, boolean ignorePilot) {
             }
             // calc MG Array here:
             if (wtype.hasFlag(WeaponType.F_MGA)) {
-                double mgaBV = 0;
-                for (Mounted possibleMG : getTotalWeaponList()) {
-                    if (possibleMG.getType().hasFlag(WeaponType.F_MG)
-                            && (possibleMG.getLocation() == weapon.getLocation())) {
-                        mgaBV += possibleMG.getType().getBV(this);
+                double mgBV = 0;
```

Comment:
This block of code appears to be repeated something like five times, perhaps we could create a utility function somewhere to reduce duplicated logic?

Reply:
Please read the description, where I addressed this.